core/src/test/scala/unit/kafka/server/OffsetDeleteRequestTest.scala

```diff
@@ -0,0 +1,165 @@
+    new ClusterConfigProperty(key = "unstable.api.versions.enable", value = "true"),
+    new ClusterConfigProperty(key = "group.coordinator.new.enable", value = "true"),
+    new ClusterConfigProperty(key = "offsets.topic.num.partitions", value = "1"),
+    new ClusterConfigProperty(key = "offsets.topic.replication.factor", value = "1")
+  ))
+  def testOffsetDeleteWithOldConsumerGroupProtocolAndNewGroupCoordinator(): Unit = {
+    testOffsetDelete(false)
+  }
+
+  @ClusterTest(clusterType = Type.ALL, serverProperties = Array(
+    new ClusterConfigProperty(key = "unstable.api.versions.enable", value = "false"),
+    new ClusterConfigProperty(key = "group.coordinator.new.enable", value = "false"),
+    new ClusterConfigProperty(key = "offsets.topic.num.partitions", value = "1"),
+    new ClusterConfigProperty(key = "offsets.topic.replication.factor", value = "1")
+  ))
+  def testOffsetDeleteWithOldConsumerGroupProtocolAndOldGroupCoordinator(): Unit = {
+    testOffsetDelete(false)
+  }
+
+  private def testOffsetDelete(useNewProtocol: Boolean): Unit = {
+    if (useNewProtocol && !isNewGroupCoordinatorEnabled) {
+      fail("Cannot use the new protocol with the old group coordinator.")
+    }
+
+    // Creates the __consumer_offsets topics because it won't be created automatically
+    // in this test because it does not use FindCoordinator API.
+    createOffsetsTopic()
+
+    // Create the topic.
+    createTopic(
+      topic = "foo",
+      numPartitions = 3
+    )
+
+    for (version <- ApiKeys.OFFSET_DELETE.oldestVersion() to ApiKeys.OFFSET_DELETE.latestVersion(isUnstableApiEnabled)) {
+
+      // Join the consumer group. Note that we don't heartbeat here so we must use
+      // a session long enough for the duration of the test.
+      val (memberId, memberEpoch) = joinConsumerGroup(
+        groupId = "grp",
+        useNewProtocol = useNewProtocol
+      )
+
+      // Commit offsets.
+      for (partitionId <- 0 to 2) {
+        commitOffset(
+          groupId = "grp",
+          memberId = memberId,
+          memberEpoch = memberEpoch,
+          topic = "foo",
+          partition = partitionId,
+          offset = 100L + partitionId,
+          expectedError = Errors.NONE,
+          version = ApiKeys.OFFSET_COMMIT.latestVersion(isUnstableApiEnabled)
+        )
+      }
+
+      // Delete offset with topic that the group is subscribed to.
+      deleteOffset(
+        groupId = "grp",
+        topic = "foo",
+        partition = 0,
+        expectedPartitionError = Errors.GROUP_SUBSCRIBED_TO_TOPIC,
+        version = version.toShort
+      )
+
+      // Unsubscribe the topic.
+      if (useNewProtocol) {
+        consumerGroupHeartbeat(
+          groupId = "grp",
+          memberId = memberId,
+          memberEpoch = memberEpoch,
+          subscribedTopicNames = List.empty
+        )
+      } else {
+        leaveGroup(
+          groupId = "grp",
+          memberId = memberId,
+          useNewProtocol = false
+        )
```

Comment:
I wonder we could also test the "unsubscribe" case for the old protocol and the leave group case for the new one for completeness.

Reply:
I don't know how to unsubscribe a topic in the old protocol except making all members subscribing to that topic leave..